### DISCUSSION TITLE
Any reason for using tuple structs instead of normal structs?

### DISCUSSION THREAD

**Discussion by mainrs** [zesterer/chumsky#279] [General]
For example here:

https://github.com/zesterer/chumsky/blob/40fe7d1966f375b3c676d01e04c5dca08f7615ac/src/combinator.rs#L464-L466

This requires to do further digging and taking a look at some methods of `Repeated` to understand what the fields are supposed to mean. Is there any reason why you did not choose to define `Repeated` as:

```rust
#[must_use]
#[derive(Copy, Clone)]
pub struct Repeated<A> {
    inner: A,
    min: usize,
    max: Option<usize>,
}
```

**Comment by zesterer** [comment]
No specific reason other than convenience at the time. `zero-copy` (which is the branch I'd recommend for all new development, given that it's a near-total rewrite of the crate) has [explicit fields](https://github.com/zesterer/chumsky/blob/zero-copy/src/zero_copy/combinator.rs#L885-L890).

That said, the fields of `Repeated` are not public deliberately and understanding them shouldn't be required to use chumsky.

**Reply by mainrs** [reply]
> That said, the fields of `Repeated` are not public deliberately and understanding them shouldn't be required to use chumsky.

Definitely not, no. I was just taking a look at `chumsky` and its API to get some inspiration for a library I am working on. That's why I noticed it.

Thanks for the explanation!

**Reply by zesterer** [reply]
Good luck with your library!
